lib/prelude.dx

```diff
@@ -664,6 +728,11 @@ def eye [Eq n] : n=>n=>Float =
   for i j. select (i == j) 1.0 0.0
 
 '## Pseudorandom number generator utilities
+Dex does not use a stateful random number generator.
+Rather it uses what is known as a split-able random number generator, which is based on a hash function.
+Dex's PRNG system is modelled directly after [JAX's](https://github.com/google/jax/blob/master/design_notes/prng.md), which is based on a well established but shockingly underused idea from the functional programming community: the splittable PRNG. It's a good idea for many reasons, but it's especially helpful in a parallel setting. If you want to read more, [Splittable pseudorandom number generators using cryptographic hashing](http://publications.lib.chalmers.se/records/fulltext/183348/local_183348.pdf) describes the splitting model itself and [D.E. Shaw Research's counter-based PRNG](http://www.thesalmons.org/john/random123/papers/random123sc11.pdf) proposes the particular hash function we use.
```

Comment:
This is literally copy pasted from @dougalm's reply in
https://github.com/google-research/dex-lang/issues/25#issuecomment-559635327